src/main/java/com/vladbudan/restapp/servlet/MainServlet.java

```diff
@@ -0,0 +1,516 @@
+            String id = request.getParameter("id");
+            if (action == null) {
+                if (id == null) {
+                    getAllUsers(request, response);
+                } else {
+                    getUser(request, response);
+                }
+            } else if (action.endsWith("/cats")) {
+                if (id == null) {
+                    getAllCats(request, response);
+                } else {
+                    getCat(request, response);
+                }
+            } else if (action.endsWith("/dogs")) {
+                if (id == null) {
+                    getAllDogs(request, response);
+                } else {
+                    getDog(request, response);
+                }
+            } else if (action.endsWith("/pets")) {
+                getAllPets(request, response);
+            }
+
+        } catch (IOException e) {
+            throw new MyException(e.getMessage());
+        }
+
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws MyException {
+
+        try {
+            String action = request.getPathInfo();
+            if (action == null) {
+                createUser(request, response);
+            } else if (action.endsWith("/cats")) {
+                createCat(request, response);
+            } else if (action.endsWith("/dogs")) {
+                createDog(request, response);
+            }
+        } catch (IOException | ParseException e) {
+            throw new MyException(e.getMessage());
+        }
+
+    }
+
+    @Override
+    protected void doPut(HttpServletRequest request, HttpServletResponse response) throws MyException {
+
+        try {
+            String action = request.getPathInfo();
+            if (action == null) {
+                updateUser(request, response);
+            } else if (action.endsWith("/cats")) {
+                updateCat(request, response);
+            } else if (action.endsWith("/dogs")) {
+                updateDog(request, response);
+            }
+        } catch (IOException | ParseException e) {
+            throw new MyException(e.getMessage());
+        }
+
+    }
+
+    @Override
+    protected void doDelete(HttpServletRequest request, HttpServletResponse response) throws MyException {
+
+        try {
+            String action = request.getPathInfo();
+            if (action == null) {
+                deleteUser(request, response);
+            } else if (action.endsWith("/cats")) {
+                deleteCat(request, response);
+            } else if (action.endsWith("/dogs")) {
+                deleteDog(request, response);
+            }
+
+        } catch (IOException e) {
+            throw new MyException(e.getMessage());
+        }
+
+    }
+
+    private void getUser(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+        int userId = Integer.parseInt(request.getParameter("id"));
```

Comment:
Why do you use here 'request.getParameter("id")
How do use send request? /users?id=1 ??? It's wrong